chromium/rules.js

```diff
@@ -15,6 +15,17 @@ const trivial_rule_from_c = new RegExp("^http:");
 const trivial_cookie_name_c = new RegExp(".*");
 const trivial_cookie_host_c = new RegExp(".*");
 
+// Empty iterable singleton to reduce memory usage (#12494).
+const nullIterator = {
+  next : () => ({ done: true })
+};
+
+const nullIterable = {
```

Comment:
Simply replace this entire code with `const nullIterable = (function* () {})()`.

Reply:
I choose to create `Object.creat(null, ...` because the created object wouldn't have extra properties, so it would be harder to misuse. This seemed important because we are using it alongside real data.

Reply:
Also, using `Object.create(...` lets us specify other properties we might want, like `size` in this case.

Reply:
Would simply doing `{ size: 0 }` cause a conflict?

Reply:
Ah, it makes it read only.